internal/controller/providers/psmdb/applier.go

```diff
@@ -694,3 +694,42 @@ func (p *applier) genPSMDBBackupSpec() (psmdbv1.BackupSpec, error) {
 
 	return psmdbBackupSpec, nil
 }
+
+func (p *applier) getCurrentReplSet(name string) *psmdbv1.ReplsetSpec {
+	for _, replset := range p.currentPSMDBSpec.Replsets {
+		if replset.Name == name {
+			return replset
+		}
+	}
+	return nil
+}
+
+func configureStorage(
+	ctx context.Context,
+	c client.Client,
+	desired *psmdbv1.ReplsetSpec,
+	current *psmdbv1.ReplsetSpec,
+	db *everestv1alpha1.DatabaseCluster,
+) error {
+	var currentSize resource.Quantity
+	if current != nil {
+		currentSize = current.VolumeSpec.PersistentVolumeClaim.PersistentVolumeClaimSpec.Resources.Requests[corev1.ResourceStorage]
```

Comment:
`.VolumeSpec.PersistentVolumeClaim.PersistentVolumeClaimSpec` technically can be null. not sure what is the chance, but the spec allows it. should we handle that case?

Reply:
The `setStorageSize` lamda function below is the only place we are setting the VolumeSpec field, and there we always initialize the `.VolumeSpec.PersistentVolumeClaim.PersistentVolumeClaimSpec` object. So at least right now, we will not get the nil-ptr deref.

But it is still good to add the additional nil-checks for future proofing. Updated, thanks :)